src/gpu/amd/miopen_binary.hpp

```diff
@@ -0,0 +1,127 @@
+
+#ifndef GPU_AMD_MIOPEN_BINARY_HPP
+#define GPU_AMD_MIOPEN_BINARY_HPP
+
+#include "common/binary_pd.hpp"
+#include "common/c_types_map.hpp"
+#include "common/primitive.hpp"
+#include "gpu/amd/miopen_binary_impl.hpp"
+#include "gpu/amd/sycl_hip_engine.hpp"
+#include "gpu/amd/sycl_hip_utils.hpp"
+#include <CL/sycl.hpp>
+#include <miopen/miopen.h>
+
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace amd {
+
+struct miopen_binary_t : public primitive_t {
+    using primitive_t::primitive_t;
+
+    struct pd_t : public binary_pd_t {
+        using binary_pd_t::binary_pd_t;
+
+        DECLARE_COMMON_PD_T("hip:miopen:any", miopen_binary_t);
+
+        status_t init(engine_t *) {
+            using namespace data_type;
+
+            bool ok = (set_default_params() == status::success)
+                    && check_data_types() && check_no_blocking()
+                    && check_broadcast()
+                    && attr()->has_default_values(
+                            primitive_attr_t::skip_mask_t::scales)
+                    && IMPLICATION(!attr()->scales_.has_default_values(),
+                            check_scales_mask());
+
+            if (!ok) return status::unimplemented;
+
+            if (check_for_zero_dims()) return status::success;
+
+            binary_impl_.reset(new miopen_binary_impl_t());
+
+            return binary_impl_->init(this);
+        }
+
+        bool check_for_zero_dims() const {
+            return has_zero_dims(src_md(0)->dims, src_md(0)->ndims)
+                    || has_zero_dims(src_md(1)->dims, src_md(1)->ndims)
+                    || has_zero_dims(dst_md()->dims, dst_md()->ndims);
+        }
+
+        bool check_scales_mask() const {
+            for (const auto &s : attr()->scales_.scales_) {
+                if (s.second.mask_ != 0) return false;
+            }
+            return true;
+        }
+
+        bool check_no_blocking() const {
+            // Blocking is not supported by MIOPENOpTensor, return false if any
+            // blocks are present
+            return src_md(0)->format_desc.blocking.inner_nblks
+                    + src_md(1)->format_desc.blocking.inner_nblks
+                    + dst_md()->format_desc.blocking.inner_nblks
+                    == 0;
+        }
+
+        bool check_broadcast() const {
+            // Source 0 broadcast is not supported
+            const int ndims = nstl::min(src_md(0)->ndims, src_md(1)->ndims);
+            for (int dim_idx = 0; dim_idx < ndims; dim_idx++) {
+                if (src_md(0)->dims[dim_idx] == 1
+                        && src_md(0)->dims[dim_idx] != src_md(1)->dims[dim_idx])
+                    return false;
+            }
+            return true;
+        }
+
+        bool check_data_types() const {
+            using namespace data_type;
+            bool inputs_same = src_md(0)->data_type == src_md(1)->data_type;
+            dnnl_data_type_t input_type = src_md(0)->data_type;
```

Comment:
Please use internal C++ API. Here is the mapping: https://github.com/oneapi-src/oneDNN/blob/master/src/common/c_types_map.hpp 
```suggestion
            data_type_t input_type = src_md(0)->data_type;
```

Reply:
Thank you for the input, we used that internal C++ API .
With reference to that  mapping we modified our implementation accordingly.